### PR TITLE
Introduce legacy s2i module to enable current s2i workflow

### PR DESCRIPTION
**Pull request by jfdenise**
* A cekit module that captures legacy s2i workflow (galleon provisioning, copy of content into server, calling extensions).
* Clean-up s2i cekit module from WildFly specific setup, allow to be shared with other technologies.

### CODE REVIEW

**Comment by jfdenise** [comment]
We can't run the checks, will be run from wildfly-s2i PR.

File: jboss/container/wildfly/s2i/bash/artifacts/opt/jboss/container/wildfly/s2i/assemble.sh [hunk 1]
```diff
@@ -8,6 +8,5 @@ source "${JBOSS_CONTAINER_WILDFLY_S2I_MODULE}/s2i-core-hooks"
 # inject our overridden maven_s2i_*() functions
 source "${JBOSS_CONTAINER_WILDFLY_S2I_MODULE}/maven-s2i-overrides"
 
-export MAVEN_ARGS_APPEND="${MAVEN_ARGS_APPEND} -Dcom.redhat.xpaas.repo.jbossorg"
```
Comment (jmesnil) [inline jboss/container/wildfly/s2i/bash/artifacts/opt/jboss/container/wildfly/s2i/assemble.sh:11]: isn't `-Dcom.redhat.xpaas.repo.jbossorg` a magic property that adds Red Hat Maven repository before `mvn package` is invoked?

File: jboss/container/wildfly/s2i/legacy/artifacts/opt/jboss/container/wildfly/s2i/assemble.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+     exit 1
+   fi
+fi
+if [ -n "${GALLEON_PROVISION_FEATURE_PACKS}" ]; then
+   if [ -z "${GALLEON_PROVISION_LAYERS}" ]; then
+     log_error "GALLEON_PROVISION_LAYERS env variable must be set when GALLEON_PROVISION_FEATURE_PACKS is set"
+     exit 1
+   fi
+fi
+if [ -n "${GALLEON_PROVISION_FEATURE_PACKS}" ]; then
```
Comment (jmesnil) [inline jboss/container/wildfly/s2i/legacy/artifacts/opt/jboss/container/wildfly/s2i/assemble.sh:18]: could we add a message that these 2 env vars are deprecated and should be replaced by the usage of the plugin?
So we can have the opportunity to actually remove them in later versions at some point

File: jboss/container/wildfly/s2i/legacy/artifacts/opt/jboss/container/wildfly/s2i/galleon/provisioning/generic_layers/pom.xml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                <artifactId>galleon-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>wildfly-provisioning-layers ${env.GALLEON_PROVISION_LAYERS}</id>
+                        <goals>
+                            <goal>provision</goal>
+                        </goals>
+                        <phase>compile</phase>
+                        <configuration>
+                            <install-dir>${project.build.directory}/server</install-dir>
```
Comment (jmesnil) [inline jboss/container/wildfly/s2i/legacy/artifacts/opt/jboss/container/wildfly/s2i/galleon/provisioning/generic_layers/pom.xml:53]: I'm a bit lost.

The installation dir for the server is hard-coded to `target/server` from this "temp" maven project which is then copied to `$JBOSS_HOME` later on by the script `galleon_replace_server`

but if we are using the `wildfly-maven-plugin` the server is installed in the `target/server` of the user's application pom.xml. Are we moving the server provisioned by the plugin to `$JBOSS_HOME` in another cekit module?

File: jboss/container/wildfly/s2i/bash/artifacts/opt/jboss/container/wildfly/s2i/assemble.sh (hunk 1 above)
Comment (jfdenise) [reply jboss/container/wildfly/s2i/bash/artifacts/opt/jboss/container/wildfly/s2i/assemble.sh:11]: Yes, it has been move to wildfly-s2i: https://github.com/wildfly/wildfly-s2i/pull/226/files#diff-61e961b6eadd77126d79b1c119b6c98dcd2b727bf14263db93ddfe4ee2c7a7d9R3

File: jboss/container/wildfly/s2i/legacy/artifacts/opt/jboss/container/wildfly/s2i/assemble.sh (hunk 2 above)
Comment (jfdenise) [reply jboss/container/wildfly/s2i/legacy/artifacts/opt/jboss/container/wildfly/s2i/assemble.sh:18]: +1

File: jboss/container/wildfly/s2i/legacy/artifacts/opt/jboss/container/wildfly/s2i/galleon/provisioning/generic_layers/pom.xml (hunk 3 above)
Comment (jfdenise) [reply jboss/container/wildfly/s2i/legacy/artifacts/opt/jboss/container/wildfly/s2i/galleon/provisioning/generic_layers/pom.xml:53]: The usage of wildfly-maven-plugin is not in this context, yes handled by the s2i module: https://github.com/wildfly/wildfly-cekit-modules/blob/v2/jboss/container/wildfly/s2i/bash/artifacts/opt/jboss/container/wildfly/s2i/s2i-core-hooks

**Pull request merged**